frontend/src/model/beta/reviews/ApprovalsDisplay.tsx

```diff
@@ -0,0 +1,50 @@
+import { Done } from '@mui/icons-material'
+import { Stack, Tooltip, Typography } from '@mui/material'
+import { useGetModelRoles } from 'actions/model'
+import { useMemo } from 'react'
+import Loading from 'src/common/Loading'
+import MessageAlert from 'src/MessageAlert'
+import { ReviewResponseWithRole } from 'types/interfaces'
+import { plural } from 'utils/stringUtils'
+
+interface ApprovalsDisplayProps {
+  modelId: string
+  acceptedReviewResponses: ReviewResponseWithRole[]
+}
+
+export default function ApprovalsDisplay({ modelId, acceptedReviewResponses }: ApprovalsDisplayProps) {
+  const { modelRoles, isModelRolesLoading, isModelRolesError } = useGetModelRoles(modelId)
+
+  const roleApprovals = useMemo(() => {
+    const staticRoles = ['owner', 'contributor', 'consumer']
+    const dynamicRoles = modelRoles.filter((role) => !staticRoles.includes(role.id)).map((role) => role)
+    return dynamicRoles.map((dynamicRole) => {
+      const acceptedResponsesForRole = acceptedReviewResponses.filter(
+        (acceptedResponse) => acceptedResponse.role === dynamicRole.id,
+      )
+      if (acceptedResponsesForRole.length > 0) {
+        return (
+          <>
```

Comment:
I've also removed this from the code snippet in my other comment, but this fragment isn't needed.